src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -0,0 +1,657 @@
+    }
+
+    /**
+     * Creates and returns a Form instance from the type of the form.
+     *
+     * @param string $type    The fully qualified class name of the form type
+     * @param mixed  $data    The initial data for the form
+     * @param array  $options Options for the form
+     *
+     * @return Form
+     */
+    protected function createForm($type, $data = null, array $options = array())
+    {
+        $this->populateFormFactory('createForm');
+
+        return $this->formFactory->create($type, $data, $options);
+    }
+
+    /**
+     * Creates and returns a form builder instance.
+     *
+     * @param mixed $data    The initial data for the form
+     * @param array $options Options for the form
+     *
+     * @return FormBuilder
+     */
+    protected function createFormBuilder($data = null, array $options = array())
+    {
+        $this->populateFormFactory('createFormBuilder');
+
+        return $this->formFactory->createBuilder(FormType::class, $data, $options);
+    }
+
+    /**
+     * Shortcut to return the Doctrine Registry service.
+     *
+     * @return Registry
+     *
+     * @throws \LogicException If DoctrineBundle is not available
+     */
+    protected function getDoctrine()
+    {
+        if (null === $this->doctrine) {
+            if (!isset($this->container) || !$this->container->has('doctrine')) {
+                throw new \LogicException('You cannot use the "getDoctrine" method if the "setDoctrine" method has not been called or if the "doctrine" service is not available.');
+            }
+
+            $this->doctrine = $this->container->get('doctrine');
+        }
+
+        return $this->doctrine;
+    }
+
+    /**
+     * Get a user from the Security Token Storage.
+     *
+     * @return mixed
+     *
+     * @throws \LogicException If SecurityBundle is not available
+     *
+     * @see TokenInterface::getUser()
+     */
+    protected function getUser()
+    {
+        if (null === $this->tokenStorage) {
+            if (!isset($this->container) || !$this->container->has('security.token_storage')) {
+                throw new \LogicException('You cannot use the "getUser" method if the "setTokenStorage" method has not been called or if the "security.token_storage" service is not available.');
+            }
+
+            $this->tokenStorage = $this->container->get('security.token_storage');
+        }
+
+        if (null === $token = $this->tokenStorage->getToken()) {
+            return;
+        }
+
+        if (!is_object($user = $token->getUser())) {
+            // e.g. anonymous authentication
+            return;
+        }
+
+        return $user;
+    }
+
+    /**
+     * Checks the validity of a CSRF token.
+     *
+     * @param string $id    The id used when generating the token
+     * @param string $token The actual token sent with the request that should be validated
+     *
+     * @return bool
+     */
+    protected function isCsrfTokenValid($id, $token)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $id. Configured minimum length is 3.](https://www.codacy.com/app/git_9/symfony/file/2331978825/issues/source?bid=3207821&fileBranchId=3207858#l645)